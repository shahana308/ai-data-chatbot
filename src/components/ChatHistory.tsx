import React from "react";

const ChatHistory = () => {
  return (
    <div className="w-2/5 bg-white p-4 h-screen overflow-y-auto">
      Chat History
    </div>
  );
};

export default ChatHistory;
