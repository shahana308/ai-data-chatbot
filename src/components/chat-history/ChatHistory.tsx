"use client";

import { useState } from "react";
import useChatStore from "stores/useChatStore";
import NewChatModal from "./NewChatModal";
import NewChatButton from "./NewChatButton";
import Header from "components/shared/Header";
import { Divider, Input } from "antd";
import ChatHistoryCard from "./ChatHistoryCard";
import { PlusOutlined } from "@ant-design/icons";

const ChatHistory = () => {
  const { addChat, chats } = useChatStore();

  const [isModalVisible, setIsModalVisible] = useState(false);
  const [newChatTitle, setNewChatTitle] = useState("");
  const [search, setSearch] = useState("");

  const handleOk = () => {
    if (newChatTitle.trim()) {
      addChat({ id: Date.now().toString(), title: newChatTitle, messages: [] });
      setNewChatTitle("");
      setIsModalVisible(false);
    }
  };

  const handleCancel = () => {
    setIsModalVisible(false);
  };

  return (
    <div className="flex flex-col bg-white my-5 rounded-3xl overflow-y-auto relative w-full md:w-1/3 lg:w-1/4 bg-white overflow-y-auto">
      <div className="p-4">
        <div className="flex justify-between items-center">
          <Header title="Chat History" />
          <PlusOutlined
            className="lg:!hidden"
            onClick={() => setIsModalVisible(true)}
          />
          <Input
            placeholder="Search Chat Groups"
            className="!w-[150px] lg:!hidden"
            onChange={(e) => setSearch(e.target.value)}
          />
        </div>

        <Divider className="!my-0" />

        <div className="hidden lg:block">
          <Input
            placeholder="Search Chat Groups"
            className="!my-4 "
            onChange={(e) => setSearch(e.target.value)}
          />

          <Divider className="!my-0" />
        </div>
      </div>

      <div className="overflow-y-scroll h-[90vh]">
        <div className="flex-grow p-2 space-y-2">
          {chats
            .filter((chat) =>
              chat.title.toLowerCase().includes(search.toLowerCase())
            )
            .map((chat) => (
              <ChatHistoryCard key={chat.id} chat={chat} />
            ))}
        </div>
        <NewChatButton onClick={() => setIsModalVisible(true)} />
      </div>

      <NewChatModal
        isVisible={isModalVisible}
        onOk={handleOk}
        onCancel={handleCancel}
        chatTitle={newChatTitle}
        setChatTitle={setNewChatTitle}
      />
    </div>
  );
};

export default ChatHistory;
