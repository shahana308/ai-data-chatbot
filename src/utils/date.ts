import dayjs from "dayjs";
import relativeTime from "dayjs/plugin/relativeTime";

dayjs.extend(relativeTime);

export const formatDateToRelativeTime = (date: dayjs.Dayjs) => {
  const now = dayjs();
  const diffInHours = now.diff(date, "hours");
  if (diffInHours > 2) {
    return date.format("DD/MM/YYYY, hh:mm A");
  }
  return date.fromNow();
};
