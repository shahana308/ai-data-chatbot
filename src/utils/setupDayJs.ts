import MockDate from "mockdate";

// Jest mock for dayjs with timezone plugin
jest.mock("dayjs", () => {
  const actualDayjs = jest.requireActual("dayjs");
  actualDayjs.extend(jest.requireActual("dayjs/plugin/utc"));
  actualDayjs.extend(jest.requireActual("dayjs/plugin/timezone"));
  actualDayjs.extend(jest.requireActual("dayjs/plugin/advancedFormat"));
  return actualDayjs;
});

// Mock the current date and time
beforeAll(() => {
  MockDate.set("2024-02-01");
});

// Reset the original date implementation
afterAll(() => {
  MockDate.reset();
});
