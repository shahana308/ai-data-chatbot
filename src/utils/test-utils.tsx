import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { render as libRender, RenderOptions } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { App, ConfigProvider } from "antd";
import { ReactElement } from "react";

// eslint-disable-next-line @typescript-eslint/no-require-imports
jest.mock("next/router", () => require("next-router-mock"));

Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});
export const queryClient = new QueryClient();

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) =>
  libRender(
    <QueryClientProvider client={queryClient}>
      <ConfigProvider theme={{ hashed: false }}>
        <App>{ui}</App>
      </ConfigProvider>
    </QueryClientProvider>,
    { ...options }
  );

// If we need to include providers use this instead
// render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
import "@testing-library/jest-dom";
export { userEvent };

export { customRender as render };
