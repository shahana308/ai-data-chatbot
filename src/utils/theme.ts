import { ThemeConfig } from "antd";

const theme: ThemeConfig = {
  token: {
    fontFamily: "Poppins",
    controlHeight: 32,
    fontSize: 14,
  },
  components: {
    Menu: {
      itemSelectedBg: "#000000",
      iconSize: 14,
      fontWeightStrong: 400,
      darkItemSelectedBg: "#000000",
      darkItemBg: "#000000",
      darkPopupBg: "#000000",
    },
    Select: {
      colorTextPlaceholder: "#000000",
    },
    Collapse: {
      padding: 0,
      contentPadding: "0 8px",
    },
    Tabs: {
      inkBarColor: "#000000",
      itemSelectedColor: "#000000",
      itemHoverColor: "#000000",
    },
    DatePicker: {
      colorPrimary: "#000000",
    },
    Progress: {
      defaultColor: "#000000",
    },
    Button: {
      colorLink: "#000000",
      colorLinkActive: "#000000",
      colorLinkHover: "#000000",
    },
    Badge: {
      colorError: "#000000",
    },
    Tag: {
      defaultBg: "#000000",
      defaultColor: "#fff",
      colorBorder: "#000000",
    },
    Carousel: {
      arrowSize: 32,
      colorBgContainer: "#000000",
    },
  },
};

export default theme;
