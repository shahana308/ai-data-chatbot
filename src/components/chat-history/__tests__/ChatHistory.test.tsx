import { render } from "utils/test-utils";
import ChatHistory from "../ChatHistory";

describe("ChatHistory", () => {
  it("renders correctly and matches snapshot", () => {
    const { container } = render(<ChatHistory />);
    expect(container).toMatchSnapshot();
  });
});
