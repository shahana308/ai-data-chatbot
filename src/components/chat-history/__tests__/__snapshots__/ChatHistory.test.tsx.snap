// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatHistory renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="ant-app"
  >
    <div
      class="flex flex-col w-1/4 bg-white p-4 my-5 rounded-3xl overflow-y-auto relative"
    >
      <div
        class="sticky top-0 z-10 p-2 bg-white border-b border-gray-200 "
      >
        <h4
          class="ant-typography #000"
        >
          Chat History
        </h4>
      </div>
      <div
        class="flex-grow overflow-y-auto p-2 space-y-2 h-[90vh]"
      />
      <div
        class="sticky bottom-0 z-10 p-2 bg-white flex justify-center"
      >
        <button
          class="ant-btn ant-btn-default !px-10 !bg-peach-200 !text-black !font-semibold"
          type="button"
        >
          <span>
            + New Chat
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
